### ISSUE TITLE
Issues with like comparisons and parameterized concatenation

### ISSUE THREAD

**Issue by warent** [kyleconroy/sqlc#700]
Take the following queries:

```sql
-- name: Test :one
select * from Demo
where txt ~~ '%' || @val || '%';

-- name: Test2 :one
select * from Demo
where txt like '%' || @val || '%';

-- name: Test3 :one
select * from Demo
where txt like concat('%', @val, '%');
```

The first one works fine.

The second one throws an error `no column reference found` which doesn't make sense because `~~` and `LIKE` are synonymous. 

The third one does not throw an error, but it does generate broken code:

```go
type Test3Params struct {
	Val interface{} `json:"val"`
	Val interface{} `json:"val"`
	Val interface{} `json:"val"`
}
```

Repo: https://github.com/warent/sqlc-bugs/tree/master/concat

**Comment by warent** [comment]
Note: same problems for postgresql-specific `ILIKE`

**Comment by kyleconroy** [comment]
This was fixed in #577, but sadly hasn't been released yet. You can use the latest tip builds. The next release won't be until MySQL is stable. However, your example had another issue, which has been fixed here: https://github.com/kyleconroy/sqlc/pull/701

**Comment by warent** [comment]
Awesome, thanks!
